cmd/zapi/cmd/post/pcap.go

```diff
@@ -102,19 +110,33 @@ loop:
 		return nil
 	}
 	if err == nil {
+		c.printStats()
 		fmt.Printf("%s: pcap posted\n", file)
 	}
 	return err
 }
 
 func (c *PcapCommand) Display(w io.Writer) bool {
-	total := atomic.LoadInt64(&c.bytesTotal)
+	total := atomic.LoadInt64(&c.pcapBytesTotal)
 	if total == 0 {
 		io.WriteString(w, "posting...\n")
 		return true
 	}
-	read := atomic.LoadInt64(&c.bytesRead)
+	read := atomic.LoadInt64(&c.pcapBytesRead)
 	percent := float64(read) / float64(total) * 100
 	fmt.Fprintf(w, "%5.1f%% %s/%s\n", percent, format.Bytes(read), format.Bytes(total))
 	return true
 }
+
+func (c *PcapCommand) printStats() {
+	if c.stats {
+		w := tabwriter.NewWriter(os.Stderr, 0, 0, 1, ' ', 0)
+		// truncate bytes written for tests
+		rbw := c.lastStatus.RecordBytesWritten
+		rbw = (rbw / 100) * 100
```

Comment:
Do you understand why this is necessary (i.e., why c.lastStatus.RecordBytesWritten varies between runs)?

Reply:
It's because zeek uids can vary in size.

Reply:
I added this for deterministic Zeek UIDs.
```diff
--- a/ztests/suite/zqd/archivestore/pcappost-stats.yaml
+++ b/ztests/suite/zqd/archivestore/pcappost-stats.yaml
@@ -1,4 +1,6 @@
 script: |
+  echo 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 > random.seed
+  export ZEEK_SEED_FILE=$PWD/random.seed
   source services.sh
   zapi -h $ZQD_HOST new -k archivestore testsp
   zapi -h $ZQD_HOST -s testsp pcappost -stats ng.pcap
```

Unfortunately, size still varies because stat.log varies. Sad.

Reply:
😞